src/Paket/Program.fs

```diff
@@ -49,6 +49,9 @@ type CLIArguments =
     | [<AltCommandLine("-f")>] Force
     | Hard
     | [<CustomCommandLine("nuget")>] Nuget of string
+//    | [<CustomCommandLine("gist")>] Gist of string * string
+//    | [<CustomCommandLine("github")>] Github of string * string
+    | [<CustomCommandLine("url")>] Url of string * string
     | [<CustomCommandLine("version")>] Version of string
     | [<CustomCommandLine("add-credentials")>] AddCredentials of string
```

Comment:
@kerams @forki Just came across this mention, apologies for the delay.

Separate parameters need to have separate CLI identifiers. The fact that the library selects an arbitrary parameter rather than failing at initialization is a bug. You should rename either of the two parameter identifiers.

It would be interesting to see if optional parameters could be supported in a reasonable manner. Why not make an issue under [UnionArgParser](https://github.com/nessos/UnionArgParser)?